orca-api/src/main/java/com/netflix/spinnaker/orca/api/TitusPreconfiguredJobProperties.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright 2020 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.orca.api;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import lombok.AllArgsConstructor;
+import lombok.Data;
+import lombok.EqualsAndHashCode;
+import lombok.NoArgsConstructor;
+
+@Data
+@EqualsAndHashCode(callSuper = false)
+@AllArgsConstructor
+public class TitusPreconfiguredJobProperties extends PreconfiguredJobStageProperties {
```

Comment:
This is an implementation, rather than part of the API contract. Titus and Kubernetes implements the extension points we create, rather than being part of the extension point contract.

Reply:
Not sure I follow. The way I thought about this is,  an implementation of the extensionpoint is supplying job configuration for a JobStage that is targetted for Titus/Kubernetes . This object is the contract to supply the necessary config for a Job to run on Titus.   

Extension point is just allowing the plugin dev to construct the config and they can use any of the configuration props supplied.

Reply:
You're right. I was thinking about this on the way home. My concern here is that the API feels like blurred lines: Titus is only one runtime of a preconfigured job, it isn't part of the core API of a preconfigured job within Orca.

I'm struggling because we do want people who are supplying Titus preconfigured jobs to have a clear contract, but seeing classes specific to a cloud provider in the core API module feels like a smell to me. My thought is: If we pull Titus out into its own plugin, I would expect that it might surface its own extensions where there would be a strong contract.

This is probably just a symptom of Orca having leaky / poor abstraction around cloud providers and certainly isn't something we're going to solve in this PR. What's here is good with how Orca is designed today. 💯

Reply:
Not sure I follow. The way I thought about this is,  an implementation of the extension point is supplying job configuration for a JobStage that is targeted for Titus/Kubernetes . This object is the contract to supply the necessary config for a Job to run on Titus.   

Extension point is just allowing the plugin dev to construct the config and they can use any of the configuration props supplied.

Reply:
Correct!. Orca knows more than it should w.r.to cloud providers and it did cross my mind that the orca api shouldn't be caring about the Titus constructs but given the current state of the integrations we are ending up bleeding that into api.